### ISSUE TITLE
Vite.json clobbers vite.config.ts custom https configuration

### ISSUE THREAD

**Issue by danderozier** [ElMassimo/vite_ruby#42]
[troubleshooting section]: https://vite-ruby.netlify.app/guide/troubleshooting.html

- [x] I have tried upgrading by running `bundle update vite_ruby`.
- [x] I have read the __[troubleshooting section]__ before opening an issue.

### Description 📖

[Vite's server.https option](https://vitejs.dev/config/#server-https) accepts either a boolean or an instance of https.ServerOptions. This allows users to specify their own SSL certificates for the dev server.

The Vite Ruby plugin coerces `https` to a boolean, which is then merged on top of the server configuration in vite.config.js. This overwrites any custom https config and prevents Vite Ruby users from using their own SSL certificates in dev.

<details>
<summary>Vite Ruby Info</summary>

_Run `bin/rake vite:info` and provide the output:_

```
bin/vite present?: true
vite_ruby: 1.1.0
vite_rails: 2.0.3
rails: 6.1.3
node: v15.10.0
npm: 7.5.3
yarn: 1.22.10
pnpm: 
ruby: ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-darwin20]

installed packages:
citycms@0.1.0 /Users/derozier/Sites/sagacity-media/citycms
├─┬ vite-plugin-ruby@1.0.16
│ └── vite@2.1.0 deduped
└── vite@2.1.0
```
</details>

### Configuration

<details>
<summary>vite.config.ts</summary>

```
...
export default defineConfig({
  plugins: [
    RubyPlugin(),
    vue()
  ],
  server: {
    https: {
      key: readFileSync("config/localhost-key.pem"),
      cert: readFileSync("config/localhost.pem")
    }
  }
})
```
</details>

<details>
<summary>config/vite.json</summary>

```
{
  "all": {
    "sourceCodeDir": "app/javascript",
    "watchAdditionalPaths": []
  },
  "development": {
    "autoBuild": true,
    "publicOutputDir": "vite-dev",
    "port": 3036,
    "host": "localhost"
  },
  "test": {
    "autoBuild": true,
    "publicOutputDir": "vite-test"
  }
}
```
</details>

### Logs 📜

_If not providing a reproduction:_

<details>
<summary>Output</summary>

_Run `DEBUG=vite-plugin-ruby:* bin/vite dev` or `DEBUG=vite-plugin-ruby:* bin/vite build` and provide the output:_

```
vite-plugin-ruby:config {
  vite-plugin-ruby:config   base: '/vite-dev/',
  vite-plugin-ruby:config   build: {
  vite-plugin-ruby:config     assetsDir: 'assets',
  vite-plugin-ruby:config     emptyOutDir: false,
  vite-plugin-ruby:config     outDir: '../../../public/vite-dev',
  vite-plugin-ruby:config     manifest: true,
  vite-plugin-ruby:config     rollupOptions: { input: [Object] },
  vite-plugin-ruby:config     sourcemap: false
  vite-plugin-ruby:config   },
  vite-plugin-ruby:config   root: '...',
  vite-plugin-ruby:config   server: { host: 'localhost', https: false, port: 3036, strictPort: true },
  vite-plugin-ruby:config   entrypoints: {
  ...
  vite-plugin-ruby:config   }
```
</details>

**Comment by ElMassimo** [comment]
Thanks for the detailed report! I wasn't expecting the plugin to override user-provided options.

I'll look into how Vite merges the options to ensure any setting provided by the user has higher precedence.

**Comment by ElMassimo** [comment]
@danderozier Released `vite-plugin-ruby` [1.0.17](https://github.com/ElMassimo/vite_ruby/blob/main/vite-plugin-ruby/CHANGELOG.md#1017-2021-03-19).

Try upgrading the package in `package.json`, and let me know how it goes.

Thanks again for the report 😃

**Comment by danderozier** [comment]
Works great — thanks for the super quick fix!
